3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -1,45 +1,64 @@
 'use strict';
 /*------------------------------------------------------------------------------
-1. Complete the function `requestData()` using `XMLHttpRequest` to make a 
-   request to the url passed to it as an argument. The function should return a 
-   promise. Make sure that the promise is rejected in case of HTTP or network 
-   errors.
+1. Complete the function `requestData()` using `fetch()` to make a request to 
+   the url passed to it as an argument. The function should return a promise. 
+   Make sure that the promise is rejected in case of HTTP or network errors.
 2. Notice that the function `main()` calls `requestData()`, passing it the url 
    `https://xkcd.now.sh/?comic=latest`. Try and run the code in the browser and 
    open the browser's console to inspect the data returned from the request.
 3. Next, complete the function `renderImage()` to render an image as an `<img>` 
-   element appended to the document's body, using the data returned from the 
-   API.
+   element appended to the document's body, using the data returned from the API.
 4. Complete the function `renderError()` to render any errors as an `<h1>` 
    element appended to the document's body.
-5. Test error handling, for instance, by temporarily changing the `.sh` in the 
+5. Refactor the `main()` function to use `async/await`.
+6. Test error handling, for instance, by temporarily changing the `.sh` in the 
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
 function requestData(url) {
-  // TODO return a promise using XMLHttpRequest
-  // TODO handle both HTTP and network errors
+  // TODO return a promise using `fetch()`
+  fetch(url)
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      console.log('data', data);
+      return data;
+    })
+    .catch((error) => {
+      console.log('err', error);
+    });
```

Comment:
We need to return a promise here as well. The way we want is to use the `requestData` in main to get the response and render it.